src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/PackageArgumentParserUtility.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.CommandLine.Parsing;
+using NuGet.CommandLine.XPlat.Commands.Package;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Utility
+{
+    internal static class PackageArgumentParserUtility
+    {
+        public static IReadOnlyList<PackageArgument<NuGetVersion>> ParseWithVersion(ArgumentResult result)
+        {
+            return PackageArgument<NuGetVersion>.Parse(
+                result,
+                NuGetVersion.TryParse,
+                versionString => Messages.Error_InvalidVersion(versionString),
+                versionComparer: new NuGetVersionEqualityComparer());
```

Comment:
since this class is used for packages passed on the command line, it's unlikely there will be a huge number of them, but the comparer is stateless, so there doesn't need to be a different instance for every package. In all of our existing comparers we have a static instance that can be reused to avoid additional allocations.